pkg/flock/flock_unix.go

```diff
@@ -0,0 +1,176 @@
+// +build linux solaris darwin freebsd
+
+package flock
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/semaphore"
+	"golang.org/x/sys/unix"
+)
+
+// New returns a Flock for path.
+func New(path string) (flock, error) {
+	// Make sure we can operate on the path.
+	fd, err := unix.Open(path, os.O_RDWR|os.O_CREATE, unix.S_IRUSR|unix.S_IWUSR)
```

Comment:
Should we use an unix.Access check for this rather then opening and closing?  Would that be faster? And then only create the file if it does not exists.